### PR TITLE
feat(customize): allow for non-master default branches

### PR DESCRIPTION
**Pull request by rimarsh**

### CODE REVIEW

File: framework/customize [hunk 1]
```diff
@@ -139,6 +140,16 @@ _customize_has_remote_repo () {
   return 1
 }
 
+_customize_get_default_branch () {
+  if [ "$CUSTOMIZE_DOTFILES_CUSTOM_DEFAULT_BRANCH" == '' ]; then
+    _customize_ensure_jq
+    _customize_ensure_hub
+    CUSTOMIZE_DOTFILES_CUSTOM_DEFAULT_BRANCH=$(hub api "/repos/$(_customize_github_get_user)/$DOTFILES_CUSTOM_REPO" | jq .default_branch)
```
Comment (ianwremmel) [inline framework/customize:147]: Is the idea here that, when there's no custom branch name specified in the dotfiles config, to use the default branch name of the customization repo?

Comment (ianwremmel) [reply framework/customize:147]: Does this definitely run after that repo gets created?

File: framework/customize [hunk 2]
```diff
@@ -183,8 +194,10 @@ _customize_update () {
       return 0
     fi
 
-    if [ "$( git branch | awk '/\*/ {print $2}')" != 'master' ]; then
-      log 'custom_environments is not on master'
+    local default_branch
+    default_branch="$(_customize_get_default_branch)"
```
Comment (ianwremmel) [inline framework/customize:198]: instead of using the global default branch, what about checking the customized repos default branch?

**Comment by ianwremmel** [comment]
also worth noting that hub doesn't work well in repos where `master` is not the default. I want to switch most everything to `gh` at some point, but I haven't gotten around to it.

File: framework/customize (hunk 1 above)
Comment (rimarsh) [reply framework/customize:147]: The idea is to use the default branch specified in the remote config, there is a default when you just create a repo before pushing any branches. It worked in my test, though I didn't explore extensively.

File: framework/customize (hunk 2 above)
Comment (rimarsh) [reply framework/customize:198]: That's what this does I think, it's fetching the default branch for this repo.

**Comment by rimarsh** [comment]
> also worth noting that hub doesn't work well in repos where `master` is not the default. I want to switch most everything to `gh` at some point, but I haven't gotten around to it.

Interesting, well we don't have to change the default it sets up... mostly my customizations repo got created with main which meant this bailed out all the time.

**Review by ianwremmel — APPROVED** [review]
ok, your issues make sense. Let's merge this for now and we can deal with my concerns when removing hub in favor of gh

**Pull request merged**